### PR TITLE
Fix #2284 - Add FAQs section to the app.

### LINKED ISSUE #2284: Add FAQs section
**Actual Behaviour**

There is no FAQ section in the app.

**Expected Behaviour**

There should be one.

**Would you like to work on the issue?**

Yes

@iamareebjamal Should we add this?

### PR DESCRIPTION
**Pull request by mayank8318**
Fixes #2284 

Changes:
Step 1) Create the data model.
@iamareebjamal Please review if I am on the right track here. Also, I checked the sample data for the event here - https://github.com/fossasia/open-event/tree/master/sample but I didn't find any json for the FAQs. How should I add it to the `assets` for offline use? Should I create my own sample FAQs?

### CODE REVIEW

**Comment by iamareebjamal** [comment]
No, if not present, don't show any FAQ section

**Comment by mayank8318** [comment]
@iamareebjamal But it is documented in the API docs?

**Comment by iamareebjamal** [comment]
But if only JSON is available, we can't show FAQ

**Comment by mayank8318** [comment]
Okay

**Comment by mayank8318** [comment]
Changes: Complete the FAQ functionality.

@iamareebjamal This is the response from Postman:

{
    "meta": {
        "count": 0
    },
    "links": {
        "self": "/v1/events/173/faqs"
    },
    "data": [],
    "jsonapi": {
        "version": "1.0"
    }
}

And this is the app log :
open1

@iamareebjamal Please review :)

File: android/app/src/main/java/org/fossasia/openevent/common/events/FAQDownloadEvent.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.fossasia.openevent.common.events;
+
+/**
+ * Created by mayank on 4/2/18.
+ */
+
+public class FAQDownloadEvent extends DownloadEvent {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/common/events/FAQDownloadEvent.java]: Please remove author comment

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Similarly from other areas

File: android/app/src/main/java/org/fossasia/openevent/common/events/FAQDownloadEvent.java (hunk 1 above)
Comment (mayank8318) [reply android/app/src/main/java/org/fossasia/openevent/common/events/FAQDownloadEvent.java]: Okay

**Comment by mayank8318** [comment]
@iamareebjamal I made a new event id **678** and added a faq using the api. I checked with postman and I am getting the results on querying. Also, I checked it by changing the api link in the config file and I am getting correct results.

**Comment by mayank8318** [comment]
Screenshots for the added FAQs :
image

**Comment by iamareebjamal** [comment]
Improve UI a little bit. Use cards, don't use colors too much. And make it so the cards are collapsed by default and expand when clicked. Please add a screenshot of what happens where there is no FAQ

**Comment by mayank8318** [comment]
Changes :
Made the cards collapsible.
27420260_1967770233251781_8946862106762805248_n

- FAQs need the user to be logged in. So I am opening login screen if the user is not logged in.
-  If the user presses back, appropriate message is shown
image

Empty FAQ screen :
image

@iamareebjamal Please review :)

**Comment by iamareebjamal** [comment]
Good job. Just change the radius of cards to be 2dp instead

**Comment by mayank8318** [comment]
@iamareebjamal Ok will do in an hour :)

**Comment by mayank8318** [comment]
@iamareebjamal made the changes.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**